src/bin/pgcopydb/ld_stream.c

```diff
@@ -35,6 +35,93 @@
 static bool updateStreamCounters(StreamContext *context,
 								 LogicalMessageMetadata *metadata);
 
+static bool convertFiltersToWal2JsonOptions(SourceFilters *filters,
+											PQExpBuffer commaSeperatedAddTables,
+											PQExpBuffer commaSeperatedFilterTables);
+
+/*
+ * This function checks if the table is in the s_table table in sourceDB internal catalog. If it exists and exclude-data is true
+ * then the message should be filtered out. Otherwise, the message should not be filtered out. If the table is not in the s_table
+ * table, then the message should be filtered out.
+ * @param shouldFilterOutMessage A pointer to a boolean variable indicating whether the message should be filtered out.
+ * Returns true if the function succeeded, false otherwise.
+ */
+bool
+ShouldFilterOutMessage(StreamContext *streamContext, char *nspname, char *relname,
+					   bool *shouldFilterOutMessage)
+{
+	/*
+	 * Validate nspname is not NULL or empty
+	 */
+	if (nspname == NULL || IS_EMPTY_STRING_BUFFER(nspname))
+	{
+		log_error("BUG: nspname is NULL or empty");
+		return false;
+	}
+
+	/*
+	 * Validate relname is not NULL or empty
+	 */
+	if (relname == NULL || IS_EMPTY_STRING_BUFFER(relname))
+	{
+		log_error("BUG: relname is NULL or empty");
+		return false;
+	}
+
+	/*
+	 * If no filters are set, then we don't need to filter out any messages
+	 */
+	if (streamContext->filters.type == SOURCE_FILTER_TYPE_NONE)
+	{
+		*shouldFilterOutMessage = false;
+		return true;
+	}
+
+	/*
+	 * If the filters are set, then we need to check if the message should be
+	 * filtered out. To do that, we lookup the table in our internal source
+	 * catalogs and check if the table is in the s_table table.
+	 */
+
+	SourceTable *table = (SourceTable *) calloc(1, sizeof(SourceTable));
+
+	if (table == NULL)
+	{
+		log_error(ALLOCATION_FAILED_ERROR);
+		return false;
+	}
+
+	if (!catalog_lookup_s_table_by_name(streamContext->sourceDB,
+										nspname,
+										relname,
+										table))
+	{
+		log_error("Failed to lookup for table \"%s\".\"%s\" in our "
+				  "internal source catalogs",
+				  nspname,
+				  relname);
+
+		free(table);
+		return false;
+	}
+
+	/*
+	 * If the table is in the s_table table, then we should filter out the
+	 * message if exclude-data is set to true. Otherwise, we should filter out.
+	 */
+	if (table->oid != 0)
+	{
+		*shouldFilterOutMessage = table->excludeData;
+	}
+	else
+	{
+		*shouldFilterOutMessage = true;
+	}
```

Comment:
nit: I prefer to do this, but I leave it upto you to decide which one is more readable.

```suggestion
	*shouldFilterOutMessage = table->oid != 0 && table->excludeData;
```